### PR TITLE
Working shooting

### PR DESCRIPTION
**Pull request by tdudzik**
Now we can shoot on opponent's board. When we destroy a ship adjacent fields are marked as hit.

### CODE REVIEW

File: client/src/main/java/io/github/expansionteam/battleships/gui/models/Field.java [hunk 1]
```diff
@@ -75,52 +77,52 @@ public int hashCode() {
...
+//                case OPPONENT:
+//                    return new OpponentField(position, occupied, shot);
+//                case PLAYER:
+//                    return new PlayerField(position, occupied, shot);
+//                default:
+//                    throw new AssertionError();
+//            }
+//        }
+//
+//    }
```
Comment (maciejkocur) [inline client/src/main/java/io/github/expansionteam/battleships/gui/models/Field.java]: :-1:
Whole code commented

File: client/src/main/java/io/github/expansionteam/battleships/common/events/data/Orientation.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package io.github.expansionteam.battleships.common.events.data;
+
+public enum Orientation {
```
Comment (YogAzathoth) [inline client/src/main/java/io/github/expansionteam/battleships/common/events/data/Orientation.java]: Awesome thing, we actually had the same idea

File: server/src/main/java/io/github/expansionteam/battleships/engine/Board.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public boolean appendShip(Field startingField, Orientation orientation, int leng
     public boolean shootField(int x, int y) {
         Field boardField = getFieldFromTheBoard(x, y);
         if (boardField.isHit()) {
-            return false;
+           return false;
```
Comment (YogAzathoth) [inline server/src/main/java/io/github/expansionteam/battleships/engine/Board.java]: Where is the mentioned refactoring? This line? ;)

File: client/src/main/java/io/github/expansionteam/battleships/gui/models/Field.java (hunk 1 above)
Comment (YogAzathoth) [inline client/src/main/java/io/github/expansionteam/battleships/gui/models/Field.java]: Exactly the same, will it be used in the future? If not, should it not be deleted? This is hardly considered "refactoring"

**Comment by YogAzathoth** [comment]
OK so overall I like it. 
I wonder why you seem to use FXML and create JavaFX objects in the code?
I do like to naming, and the event halding stuff.
Only thing I dislike (not in all cases) are some of the commit messages, you are basically describing what I can see in the code, a little more logic or the purpose of the changes to actually describe the commit would be nice.

**Pull request merged**